grouper/models.py

```diff
@@ -118,6 +130,17 @@ def get_db_engine(url):
     return create_engine(url, pool_recycle=300)
 
 
+Plugins = []
+
+
+def load_plugins(plugin_dir):
+    """Load plugins from a directory"""
+    global Plugins
+    if Plugins:
+        raise Exception("Plugins already loaded; can't load twice!")
```

Comment:
Is it general good practice to make lots of named classes for
exceptions? I don't know what "best" is :)

--
Mark Smith mark@qq.is


On Sat, Feb 21, 2015, at 03:33 PM, Jathan McCollum wrote:
> In grouper/models.py[1]:


> > @@ -118,6 +130,17 @@ def get_db_engine(url):
>      return create_engine(url, pool_recycle=300)
>
>
> +Plugins = [] + + +def load_plugins(plugin_dir):
> +    """Load plugins from a directory"""
> +    global Plugins
> +    if Plugins:
> +        raise Exception("Plugins already loaded; can't load twice!")
>          I feel like this should have its own exception. Something
>          like:


> raise PluginsAlreadyLoaded("Plugins already loaded; can't load
> twice!") That way you can catch that exception? Base Exception can be
> hard to isolate.


> — Reply to this email directly or view it on GitHub[2].

>



Links:

  1. https://github.com/dropbox/grouper/pull/9#discussion_r25127047
  2. https://github.com/dropbox/grouper/pull/9/files#r25127047